fix: quests from version 11 revised

htc16 opened the pull request:
# Description

- [x] 11.02 - [Festive Outfits Quest](https://tibia.fandom.com/wiki/Festive_Outfits_Quest)
- [x] 11.02 - [Forgotten Knowledge Quest](https://tibia.fandom.com/wiki/Forgotten_Knowledge_Quest)

At position (32454, 32115, 7) change port id 6251 to port id 6252
at position {x = 31978, y = 32852, z = 14} the vortex goes to {x = 31975, y = 32858, z = 15}
at position {x = 31977, y = 32861, z = 15} the vortex goes to {x = 31981, y = 32850, z = 14}
at position {x = 31978, y = 32842, z = 14} the vortex goes to {x = 31975, y = 32842, z = 15}
at position {x = 31977, y = 32845, z = 15} the vortex goes to {x = 31981, y = 32843, z = 14}
at position {x = 31994, y = 32842, z = 14} the vortex goes to {x = 31992, y = 32842, z = 15}
at position {x = 31993, y = 32845, z = 15} the vortex goes to {x = 31991, y = 32844, z = 14}
at position {x = 31994, y = 32852, z = 14} the vortex goes to {x = 31991, y = 32858, z = 15}
at position {x = 31993, y = 32861, z = 15} the vortex goes to {x = 31991, y = 32850, z = 14}

- [x] 11.02 - [The First Dragon Quest](https://tibia.fandom.com/wiki/The_First_Dragon_Quest)

In this position (33598, 30996, 12) you need to fill in item 1128, as it is obstructing the view below.
At position (33597, 30996, 14) remove the actionid that exists in this teleport as I will go up through tables.
At position (33583, 30992, 14) remove the actionid and uniqueid, as this is already set at initialization by the table.
At position (33584, 31004, 15) this line appears on the screen when you are in the lever room.


- [x] 11.40 - [Cults of Tibia Quest](https://tibia.fandom.com/wiki/Cults_of_Tibia_Quest)

At position (32723, 31582, 8) there is an error in the tile.
at position (32736, 31450, 8) has item id 837, replace with id 27866.
at position (32736, 31451, 8) has item id 837, replace with id 27866.
at position (32736, 31452, 8) has item id 840, replace with id 25739.
At position (32761, 31523, 9) replace port id 5115 with 5116
In area (33188, 31894, 15), you need to close the black side of the client screen.
at position (33131, 31920, 8) you need to fill in to not display this on the floor below.
at position (33173, 31834, 15) you need to fill in to avoid displaying a black screen on the client.
at position (33155, 31860, 15) a black line and a gray line appear on the client.

- [x] 11.40 - [Threatened Dreams Quest](https://tibia.fandom.com/wiki/Threatened_Dreams_Quest)

Delete the npc and area at position (10253, 10033, 1)
At position {x = 33829, y = 32187, z = 14} put actionid 0
At position {x = 33982, y = 32234, z = 14} put actionid 0
At position {x = 33831, y = 32138, z = 14} put actionid 0
At position {x = 34021, y = 32037, z = 14} put actionid 0
At position {x = 33784, y = 32205, z = 14} put actionid 0
At position {x = 33921, y = 32402, z = 14} put actionid 0

- [x] 11.50 - [Dangerous Depths Quest](https://tibia.fandom.com/wiki/Dangerous_Depths_Quest)
- [x] 11.50 - [Hidden Threats Quest](https://tibia.fandom.com/wiki/Hidden_Threats_Quest)
- [x] 11.50 - [Makeshift Warrior Outfits Quest](https://tibia.fandom.com/wiki/Makeshift_Warrior_Outfits_Quest)
- [x] 11.80 - [Battle Mage Outfits Quest](https://tibia.fandom.com/wiki/Battle_Mage_Outfits_Quest)
- [x] 11.80 - [Measuring Tibia Quest](https://tibia.fandom.com/wiki/Measuring_Tibia_Quest) (**You need to develop this quest**)
- [x] 11.80 - [The Secret Library Quest](https://tibia.fandom.com/wiki/The_Secret_Library_Quest) (**It is only necessary to develop part 2 of this quest.**)
- [x] Quests have been tested in development!


### _I'm currently working on adding all quests to the PR. Given the large number of quests, I would like to ask for your help and collaboration to close this PR as quickly as possible. If you are interested in helping, choose one of the quests to review, open a PR for it, and link to this PR. All help is welcome!_


Note1: my focus is on quests, therefore, the quest log for a quest may be missing as this is not shown in the quest tutorial lists. I study tutorials and release fix ones or develop something related to the tutorial.

Note2: I have limitations as this is the first time I am willing to do this, so please consider that some codes may not be the best but what matters is the intention and this can be improved in the future if someone finds it appropriate to do so.

Note3: I'm taking the opportunity to catalog all the aids and uids so that we can eventually modify and change them in the future directly through the script.

Note4: this PR needs some updates to the map that were reported above.

Note5: I was unable to place the writings in the books, only through the map editor, but this did not influence the development of the quest, so I left this for a future review.

## Behavior
### **Real**
I reviewed and corrected any quests that weren't working.

Delete options that are not relevant.

 - [x] Bug fix (rolling change that fixes a problem)
 - [x] New feature (rolling change that adds functionality)
 - [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How this was tested

As I reviewed and modified the codes, I tested them.

**Test Setup**:

 - Server version: current
 - Client: 13
 - Operating system: Windows 11

## Checklist

 - [x] My code follows the style guidelines of this project
 - [x] I performed a self-review of my own code
 - [x] I checked the PR checks reports
 - [x] I commented my code, especially in areas that were difficult to understand
 - [x] I made corresponding changes to the documentation
 - [x] My changes do not generate new warnings
 - [x] I added tests that prove my fix is ​​effective or that my feature works

htc16 committed:
Fix: ferumbras Ascendant Quest - review 10

htc16 committed:
More Fix: ferumbras Ascendant Quest - review 10

htc16 committed:
Fix: mission 02 of The Djinn War Quest, Efreet Faction (Green Djinns) - review 7

htc16 committed:
Fix: Barbarian Test - review 8

htc16 committed:
Festive Outfits Quest and The First Dragon Quest

github-actions[bot] committed:
Lua code format - (Stylua)

htc16 committed:
More Fix: ferumbras Ascendant Quest - review 10

htc16 committed:
Fix Bosslever system

Before only a normal player could pull a lever and go to the boss room even though that wasn't the minimum requirement.

github-actions[bot] committed:
Lua code format - (Stylua)

htc16 committed:
More Fix: ferumbras Ascendant Quest - review 10

I'm testing more possibilities.

github-actions[bot] committed:
Lua code format - (Stylua)

htc16 committed:
Forgotten Knowledge Quest

github-actions[bot] committed:
Lua code format - (Stylua)

murilo09 commented:
I've converted the PR to Draft while it's in development.

htc16 committed:
Cults of Tibia Quest

github-actions[bot] committed:
Lua code format - (Stylua)

htc16 committed:
Update storages.lua

htc16 committed:
Threatened Dreams Quest

htc16 committed:
Dangerous Depths Quest

github-actions[bot] committed:
Lua code format - (Stylua)

htc16 committed:
Makeshift Warrior Outfits Quest

This is already set up.

htc16 committed:
Update storages.lua

murilo09 committed:
Merge branch 'main' into Review-Quests-V12

htc16 committed:
The Secret Library Quest - part 1

github-actions[bot] committed:
Lua code format - (Stylua)

htc16 committed:
more

htc16 committed:
The Secret Library Quest - part 1

htc16 committed:
Merge branch 'Review-Quests-V12' of https://github.com/htc16/canary into Review-Quests-V11

htc16 committed:
The Secret Library Quest - more

github-actions[bot] committed:
Lua code format - (Stylua)

htc16 committed:
The Secret Library Quest - more

github-actions[bot] committed:
Lua code format - (Stylua)

htc16 committed:
The Secret Library Quest - more

github-actions[bot] committed:
Lua code format - (Stylua)

htc16 committed:
The Secret Library Quest - more

htc16 committed:
The Secret Library - quest log

github-actions[bot] committed:
Lua code format - (Stylua)

htc16 committed:
Update quests.lua

htc16 committed:
Update quests.lua

omarcopires commented on data/scripts/actions/items/foods.lua:
```diff
@@ -136,6 +136,14 @@ function food.onUse(player, item, fromPosition, target, toPosition, isHotkey)
 		return true
 	end
 
+	-- The Secret Library Quest
```
`if IsRunningGlobalDatapack() then`

htc16 committed:
@omarcopires

good

majestyotbr committed:
Merge branch 'main' into Review-Quests-V12

htc16 committed:
The Secret Library Quest - Liquid Death

github-actions[bot] committed:
Lua code format - (Stylua)

htc16 committed:
The Secret Library Quest - Bursting at the Seams

github-actions[bot] committed:
Lua code format - (Stylua)

htc16 committed:
The Secret Library Quest - The Lament

github-actions[bot] committed:
Lua code format - (Stylua)

htc16 committed:
The Secret Library Quest - The Order of the Falcon

github-actions[bot] committed:
Lua code format - (Stylua)

htc16 committed:
The Secret Library Quest - The Path of Defiances

github-actions[bot] committed:
Lua code format - (Stylua)

htc16 committed:
The Secret Library Quest - High and Dry

github-actions[bot] committed:
Lua code format - (Stylua)

htc16 committed:
The Secret Library Quest - The Veiled Hoard of Zathroth

github-actions[bot] committed:
Lua code format - (Stylua)

htc16 committed:
The Secret Library Quest - The Secret Library: Bosses - 1º Part

htc16 committed:
Merge branch 'Review-Quests-V12' of https://github.com/htc16/canary into Review-Quests-V11

htc16 committed:
The Secret Library Quest - Addons and more

htc16 commented:
> I've converted the PR to Draft while it's in development.

Convert to PR please as I am done.

murilo09 commented:
> > I've converted the PR to Draft while it's in development.
> 
> Convert to PR please as I am done.

Done!

murilo09 committed:
Merge branch 'main' into Review-Quests-V12

htc16 committed:
Update door_quest.lua

htc16 committed:
Ferumbras Ascension - Levers

github-actions[bot] committed:
Lua code format - (Stylua)

htc16 committed:
Update actions_zamulosh.lua

htc16 committed:
Merge branch 'Review-Quests-V12' of https://github.com/htc16/canary into Review-Quests-V11

htc16 committed:
Ferumbras Ascension - Levers - final

htc16 committed:
Update actions_ragiaz.lua

github-actions[bot] committed:
Lua code format - (Stylua)

htc16 committed:
Update actions_razzagorn.lua

github-actions[bot] committed:
Lua code format - (Stylua)

htc16 committed:
Forgotten Knowledge Quest - new levers

github-actions[bot] committed:
Lua code format - (Stylua)

htc16 committed:
Update actions_the_time_guardian.lua

htc16 committed:
Update actions_frozen_horror.lua

Glatharth commented:
The levers for the bosses are on an old and non-standard model, they need to be standardized to make it easier to maintain and implement future features. I'm going to make the lib available so that you can be set up with the new standard. In the file there is an example of how the “config” table works with all the resources you'll need. I will also provide some examples of levers that use the “onUseExtra” resource for extra resources.
[lib](https://github.com/opentibiabr/canary/blob/89b307336546d23bd0b72c91baa3303c9cbe0423/data/libs/functions/boss_lever.lua#L25)
[Example 1](https://github.com/opentibiabr/canary/blob/main/data-otservbr-global/scripts/actions/bosses_levers/anomaly.lua)
[Example 2](https://github.com/opentibiabr/canary/blob/main/data-otservbr-global/scripts/actions/bosses_levers/eradicator.lua)

htc16 commented:
> The levers for the bosses are on an old and non-standard model, they need to be standardized to make it easier to maintain and implement future features. I'm going to make the lib available so that you can be set up with the new standard. In the file there is an example of how the “config” table works with all the resources you'll need. I will also provide some examples of levers that use the “onUseExtra” resource for extra resources. [lib](https://github.com/opentibiabr/canary/blob/89b307336546d23bd0b72c91baa3303c9cbe0423/data/libs/functions/boss_lever.lua#L25) [Example 1](https://github.com/opentibiabr/canary/blob/main/data-otservbr-global/scripts/actions/bosses_levers/anomaly.lua) [Example 2](https://github.com/opentibiabr/canary/blob/main/data-otservbr-global/scripts/actions/bosses_levers/eradicator.lua)

At this time I will keep the levers as they are, since they have been tested and are working as expected for the quest. Nothing prevents this from being updated if necessary in the future. There has already been extensive discussion about this.

murilo09 committed:
Merge branch 'main' into Review-Quests-V12

Glatharth commented:
Fortunately, the `bossLever` system is linked to several other systems and it is extremely important that it works, although it is not implemented in all the project's code, it does not mean that it should not be implemented. I've implemented the systems, but I don't have permission to push them to your repository, but you can review them and merge them into your repository. I'm not sure if all the levers are in the default, if there are any missing, you can implement them.

[https://github.com/Glatharth/canary/tree/Review-Quests-V12](https://github.com/Glatharth/canary/tree/Review-Quests-V12)

A few points to consider:

- **Timer Storage**: `bossLever` doesn’t use storage to track time, so the `timer` storage won’t function as expected and should be adjusted. I’ve provided an example using *[Oberon](https://github.com/Glatharth/canary/blob/342c5a42ec6cf271ae3c5dfef04020bc8e9e097c/data-otservbr-global/scripts/quests/the_secret_library_quest/the_order_of_the_falcon/movements_bossEntrance.lua#L39)* as a reference if needed.

In the code files I modified, I’ve left some notes:

- `ferumbras_ascension/actions_ferumbras_lever.lua`
  - Updated the crystal IDs as they were incorrect. A question remains: are these crystals ever swapped? If so, ensure the IDs are accurate; otherwise, the loop may be redundant and should be removed.
  - Why use `globalStorage` if it’s not utilized anywhere else?

- `liquid_death/movements_bossWay.lua`
  - This script doesn't work, need to check if map has changed

If I'm wrong somewhere you can correct me, I haven't checked everything, they're just observations.

murilo09 reviewed (changes requested):
@htc16 could you check the files @Glatharth provided?

htc16 commented:
> Fortunately, the system is linked to several other systems and it is extremely important that it works, although it is not implemented in all the project's code, it does not mean that it should not be implemented. I've implemented the systems, but I don't have permission to push them to your repository, but you can review them and merge them into your repository. I'm not sure if all the levers are in the default, if there are any missing, you can implement them.`bossLever`
> 
> https://github.com/Glatharth/canary/tree/Review-Quests-V12
> 
> A few points to consider:
> 
> * **Timer Storage**:  doesn’t use storage to track time, so the  storage won’t function as expected and should be adjusted. I’ve provided an example using _[Oberon](https://github.com/Glatharth/canary/blob/342c5a42ec6cf271ae3c5dfef04020bc8e9e097c/data-otservbr-global/scripts/quests/the_secret_library_quest/the_order_of_the_falcon/movements_bossEntrance.lua#L39)_ as a reference if needed.`bossLever``timer`
> 
> In the code files I modified, I’ve left some notes:
> 
> * `ferumbras_ascension/actions_ferumbras_lever.lua`
>   
>   * Updated the crystal IDs as they were incorrect. A question remains: are these crystals ever swapped? If so, ensure the IDs are accurate; otherwise, the loop may be redundant and should be removed.
>   * Why use  if it’s not utilized anywhere else?`globalStorage`
> * `liquid_death/movements_bossWay.lua`
>   
>   * This script doesn't work, need to check if map has changed
> 
> If I'm wrong somewhere you can correct me, I haven't checked everything, they're just observations.

I will not retest everything I have already done, if you want to bring this to this PR feel free or wait for the PR to be merged and make a bosslever PR. Thank you.

@murilo09

htc16 committed:
crystal IDs - Ferumbras lever

htc16 committed:
Merge remote-tracking branch 'upstream/main' into Review-Quests-V11

majestyotbr committed:
Merge branch 'main' into Review-Quests-V12

majestyotbr commented:
Updated global map:
https://github.com/opentibiabr/canary/releases/download/v3.1.0/otservbr.otbm
